src/algebra/lie/subalgebra.lean

```diff
@@ -456,6 +456,29 @@ begin
   { rw [← coe_to_submodule_mk p h, coe_to_submodule, coe_to_submodule_eq_iff, lie_span_eq], },
 end
 
+variables (R L)
+
+/-- `lie_span` forms a Galois insertion with the coercion from `lie_subalgebra` to `set`. -/
+protected def gi : galois_insertion (lie_span R L : set L → lie_subalgebra R L) coe :=
+{ choice    := λ s _, lie_span R L s,
```

Comment:
I think it's better to use a choice function like this:

https://github.com/leanprover-community/mathlib/blob/98e8408ac78f3dedc6a2f3d2fec89a39b5ce20f0/src/algebra/algebra/subalgebra.lean#L481

Because that means that `coe (choice s)` is definitionally equal to `s`.

Reply:
I think I'll take you up on that offer to be persuaded if it's still available!

Reply:
It matches what we do with `submodule`, so I guess I have no grounds to complain:

https://github.com/leanprover-community/mathlib/blob/c9593dcc5333e7184216f2082401434745b77170/src/linear_algebra/basic.lean#L948-L953